mkosi/__init__.py

```diff
@@ -2012,6 +2012,13 @@ def invoke_dnf(
     with mount_api_vfs(args, root):
         run(cmdline)
 
+    # On Debian, rpm/dnf ship with a patch to store the rpmdb under ~/
+    # so it needs to be copied back in the right location, otherwise
+    # the rpmdb will be broken. See: https://bugs.debian.org/1004863
+    if root.joinpath("root/.rpmdb").exists():
+        unlink_try_hard(root / "var/lib/rpm")
+        shutil.move(cast(str, root / "root/.rpmdb"), root / "var/lib/rpm")
```

Comment:
I think this hack should be conditionalized on being on Debian. Other people may have `/root/.rpmdb` for other reasons, and this shouldn't matter.

The database actually lives under `/usr/lib/sysimage/rpm/` now. (In opensuse since a while, in Fedora since ~now, https://fedoraproject.org/wiki/Changes/RelocateRPMToUsr). I have no idea how to figure out the right location automatically.

Reply:
I can add more conditionals - but is it reasonable to have /root/.rpmdb in a freshly built image for reasons other than this debian thing?

Reply:
added the conditionals, and checking if the new location exists